Assets/Tests/InputSystem/CoreTests_Analytics.cs

```diff
@@ -205,5 +216,181 @@ public void TODO_Analytics_ReceivesEventOnFirstUserInteraction()
     {
         Assert.Fail();
     }
+
+    [Test]
+    [Category("Analytics")]
+    public void Analytics__ShouldReportEditorSessionAnalytics__IfAccordingToEditorSessionAnalyticsFiniteStateMachine()
+    {
+        CollectAnalytics(InputAnalytics.InputActionsEditorSessionAnalytic.kEventName);
+
+        // Editor session analytics is stateful and instantiated
+        var session = new InputAnalytics.InputActionsEditorSessionAnalytic(
+            InputAnalytics.InputActionsEditorKind.EmbeddedInProjectSettings);
+
+        session.Begin();                    // the user opens project settings and navigates to Input Actions
+        session.RegisterEditorFocusIn();    // when window opens, it receives edit focus directly
+        runtime.currentTime += 5;           // the user is just grasping what is on the screen for 5 seconds
+        session.RegisterActionMapEdit();    // the user adds an action map or renames and action map or deletes one
+        session.RegisterActionEdit();       // the user adds an action, or renames it, or deletes one or add binding
+        session.RegisterBindingEdit();      // the user modifies a binding configuration
+        session.RegisterEditorFocusOut();   // the window looses focus due to user closing e.g. project settings
+        session.End();                      // the window is destroyed and the session ends.
+
+        // Assert: Registration
+#if (UNITY_2023_2_OR_NEWER && UNITY_EDITOR)
+        // Registration is a responsibility of the framework
+        Assert.That(registeredAnalytics.Count, Is.EqualTo(0));
+#else
+        Assert.That(registeredAnalytics.Count, Is.EqualTo(1));
+        Assert.That(registeredAnalytics[0].name, Is.EqualTo(InputAnalytics.InputActionsEditorSessionAnalytic.kEventName));
+        Assert.That(registeredAnalytics[0].maxPerHour, Is.EqualTo(100));            // REVIEW: what to use?
```

Comment:
Not sure what the comment is suggesting here.
If we're not sure what this should be why not remove the asset. It doesn't seem critical to the tast. Likewise with the one below.

Reply:
I have modified these in a pending commit to only verify that the class constant is forwarded as expected via the send analytics functionality so hence asserting against kMaxEventsPerHour, kMaxNumberOfElements now. Its basically just configuration, but this proves that values are not modified along the way.